### PR TITLE
Automated cherry pick of #126449: kube-up.sh: bump coreDNS to 1.11.3

### PR DESCRIPTION
**Pull request by BenTheElder**
Cherry pick of #126449 on release-1.31.

#126449: kube-up.sh: bump coreDNS to 1.11.3

For details on the cherry pick process, see the [cherry pick requests](https://git.k8s.io/community/contributors/devel/sig-release/cherry-picks.md) page.

```release-note
upgrade coreDNS to v1.11.3
```

### CODE REVIEW

**Review by neolit123 — COMMENTED** [review]
/lgtm
/approve

**Review by saschagrunert — APPROVED** [review]
/hold

Do we want to wait for CP's into v1.30, v1.29 and v1.28?

Referencing slack discussion: https://kubernetes.slack.com/archives/CJH2GBF7Y/p1724105677231349

**Comment by BenTheElder** [comment]
> Do we want to wait for CP's into v1.30, v1.29 and v1.28?

TLDR: I don't think we need to, and I would personally recommend that we at least land this in 1.31.1 even if we miss the other branches, but I will defer to kubeadm maintainers to unhold or maintain the hold.

<details>
IMHO: This really should've shipped in 1.31 but it was available too late in the release.

For 1.31 this is a patch version bump that fixes a number of bugs and probably CVEs (just given the delta in go compiler versions alone), and should be very safe and highly desired.

For the older branches it's less clean, less clear, and there will be the option to upgrade to v1.31.1+ to pick it up.

I think the discussion for the other branches is going to take a while due to the transitive dependencies issue and less straightforward unit test patches, but they're not strictly related to 1.31.1
</details>
